portal_blocks_documentation/docs.py

```diff
@@ -81,48 +101,145 @@ def write_official_doc(doc_json_file: pathlib.Path, block_name: str):
+        doc.extend(["### Inputs", "\n".join(doc_json["inputs"])])
 
     if doc_json.get("output", False):
-        doc.extend([
-            "### Output",
-            "\n".join(doc_json["output"])
-        ])
+        doc.extend(["### Output", "\n".join(doc_json["output"])])
 
-    block_image_url = (f"https://raw.githubusercontent.com/battlefield-portal-community/"
-                       f"Image-CDN/main/portal_blocks/{block_name}.png")
+    block_image_url = (
+        f"https://raw.githubusercontent.com/battlefield-portal-community/"
+        f"Image-CDN/main/portal_blocks/{block_name}.png"
+    )
 
     doc.append(f"\n![{block_name}]({block_image_url})")
 
     doc.append(DISCLAIMER)
     doc_file = BUILD_DIR / block_name / "docs" / "official.md"
     doc_file.write_text("\n".join(doc))
 
-def create_menu_entry(block_name: str):
-    sub_lookup = '# - name: generated-block-reference-do-not-remove'
-    menu_entry_name = f'- name: {block_name}'
-    menu_entry_ref = f'  ref: "/portal-builder/rules-editor/block-reference/{block_name}"'
+
+def cleanup_menu_entries(sub_lookup_start: str, sub_lookup_end: str):
     if MENU_FILE.exists():
-        entries = MENU_FILE.read_text().split('\n')
+        startIndex = -1
+        endIndex = -1
+        entries = MENU_FILE.read_text().split("\n")
         for index, entry in enumerate(entries):
-            if sub_lookup in entry:
-                entries.insert(index, entry.split('#')[0] + menu_entry_ref)
-                entries.insert(index, entry.split('#')[0] + menu_entry_name)
+            if sub_lookup_start in entry:
+                startIndex = index
+            if sub_lookup_end in entry:
+                endIndex = index
+            if startIndex > -1 and endIndex > -1:
                 break
-        MENU_FILE.write_text('\n'.join(entries))
+        if endIndex - startIndex > 1:
+            for i in range(startIndex, endIndex-1):
+                deletedEntry = entries.pop(startIndex + 1)
+                logger.debug(f"deleted entry from menu: {deletedEntry}")
+        else:
+            logger.debug(
+                f"No menu entries to be cleaned up in defined section (start: {startIndex}, end: {endIndex}"
+            )
+        MENU_FILE.write_text("\n".join(entries))
 
-def generate():
 
+def insert_menu_entry(menu_entry_name: str, menu_entry_ref: str, insert_before_lookup: str):
+    if MENU_FILE.exists():
+        entries = MENU_FILE.read_text().split("\n")
+        for index, entry in enumerate(entries):
+            if insert_before_lookup in entry:
+                entries.insert(index, entry.split("#")[0] + menu_entry_ref)
+                entries.insert(index, entry.split("#")[0] + menu_entry_name)
+                break
+        MENU_FILE.write_text("\n".join(entries))
+
+
+def generate_maps_doc():
+    timer_start = timer()
+    logger.info("Building map layout docs...")
+    logger.debug("Read map and mode information...")
+    jsonFile = open(MAPS_JSON_FILE)
```

Comment:
please use a context manager for file handling.. see line 90.

Reply:
as this file is opened but then never closed... in context manager it is handled automatically